spv/verifymerkleproof.go

```diff
@@ -62,30 +72,33 @@ func (v *verifier) VerifyMerkleProof(ctx context.Context, proof []byte) (valid,
 		var err error
 		merkleRoot, err = bc.ExtractMerkleRootFromBlockHeader(mpb.target)
 		if err != nil {
-			return false, false, err
+			return response, err
 		}
 
 	default:
-		return false, false, errors.New("invalid flags")
+		return response, ErrInvalidMerkleFlags
 	}
 
 	if mpb.flags&proofTypeFlag == 1 {
-		return false, false, errors.New("only merkle branch supported in this version") // merkle tree proof type not supported
+		return response, ErrInvalidMerkleFlags
 	}
 
 	if mpb.flags&compositeFlag == 1 {
-		return false, false, errors.New("only single proof supported in this version") // composite proof type not supported
+		return response, ErrInvalidMerkleFlags // composite proof type not supported
 	}
 
 	if txid == "" {
-		return false, false, errors.New("txid missing")
+		return response, ErrMissingTxidInProof
 	}
 
 	if merkleRoot == "" {
-		return false, false, errors.New("merkleRoot missing")
+		return response, ErrMissingRootInProof
 	}
 
-	return verifyProof(txid, merkleRoot, mpb.index, mpb.nodes)
+	valid, isLastInTree, err := verifyProof(txid, merkleRoot, mpb.index, mpb.nodes)
+	response.Valid = valid
```

Comment:
Suggestion: I would return
```
	return &MerkleProofValidation{
		TxID:         txid,
		Valid:        valid,
		IsLastInTree: isLastInTree,
	}, err
```
here and return `nil` for the response in the cases above where we return an error

Reply:
That is what we're doing to be clear, but I'll change the layout so that it's clearer - at the moment response is MerkleProofValidation pointer